youtube2zim/scraper.py

```diff
@@ -572,71 +578,69 @@ def get_slot():
 
         return overall_succeeded, overall_failed
 
-    def download_s3_cache(self, s3_key, local_video_path):
-        """returns True if successfully downloaded cache, otherwise False"""
+    def download_from_cache(self, key, video_path):
+        """ whether it successfully downloaded from cache """
+        if self.use_any_optimized_version:
+            if not self.s3_storage.has_object(key, self.s3_storage.bucket_name):
+                return False
+        else:
+            if not self.s3_storage.has_object_matching_meta(
+                key, tag="encoder_version", value=ENCODER_VERSION
+            ):
+                return False
+        video_path.parent.mkdir(parents=True, exist_ok=True)
         try:
-            if self.use_any_optimized_version:
-                if not self.s3_storage.has_object(s3_key, self.bucket_name):
-                    return False
-            else:
-                if not self.s3_storage.has_object_matching_meta(
-                    s3_key,
-                    tag="encoding_version",
-                    value=ENCODER_VERSION,
-                    bucket_name=self.bucket_name,
-                ):
-                    return False
-            os.makedirs(os.path.dirname(local_video_path), exist_ok=True)
-            self.s3_storage.download_file(s3_key, local_video_path)
+            self.s3_storage.download_file(key, video_path)
+            logger.info(f"cache ({key}) downloaded {video_path}")
+            return True
         except Exception as ex:
```

Comment:
could you use `exc` instead of `ex` for the exception variable? Not important but that's a convention we use. applies to upload as well